store/sqlstore/team_store.go

```diff
@@ -194,41 +194,37 @@ func (s SqlTeamStore) Save(team *model.Team) store.StoreChannel {
 	})
 }
 
-func (s SqlTeamStore) Update(team *model.Team) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		team.PreUpdate()
+func (s SqlTeamStore) Update(team *model.Team) (*model.Team, *model.AppError) {
 
-		if result.Err = team.IsValid(); result.Err != nil {
-			return
-		}
+	team.PreUpdate()
 
-		oldResult, err := s.GetMaster().Get(model.Team{}, team.Id)
-		if err != nil {
-			result.Err = model.NewAppError("SqlTeamStore.Update", "store.sql_team.update.finding.app_error", nil, "id="+team.Id+", "+err.Error(), http.StatusInternalServerError)
-			return
-		}
+	if err := team.IsValid(); err != nil {
+		return team, err
```

Comment:
I would expect to `return nil, err` only in very rare and special cases we return an object and an error, we prefer to return an object or an error.

Reply:
In some rare cases, when we ignore the error, the `*model.Team` object would be updated to nil.
ex:
`
team, _ := s.Update(team)
`

I can update it to `nil`, as its preferred choice.

Reply:
Yes, is the preferred choice, if I return and error, you shouldn't be using the success value at all. If you are not handling the error you must have a good reason for that. I think the unique case where we don't handle the error is when we use it for testing (and actually I think is better to handle the error there too, normally with a `require.Nil(t, err)`).